beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/repository/HousekeepingTableRepository.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (C) 2019-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.core.repository;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import com.expediagroup.beekeeper.core.model.EntityHousekeepingTable;
+
+public interface HousekeepingTableRepository extends JpaRepository<EntityHousekeepingTable, Long> {
+
+  @Query(value = "from EntityHousekeepingTable t where t.cleanupTimestamp <= :instant "
+      + "and (t.housekeepingStatus = 'SCHEDULED' or t.housekeepingStatus = 'FAILED') "
+      + "and t.modifiedTimestamp <= :instant order by t.modifiedTimestamp")
+  Page<EntityHousekeepingTable> findRecordsForCleanupByModifiedTimestamp(@Param("instant") LocalDateTime instant,
+      Pageable pageable);
+
+  @Query(value = "select t from EntityHousekeepingTable t "
+      + "where t.databaseName = :databaseName "
+      + "and t.tableName = :tableName "
+      + "and (t.housekeepingStatus = 'SCHEDULED' or t.housekeepingStatus = 'FAILED')")
+  Optional<EntityHousekeepingTable> findRecordForCleanupByDatabaseAndTable(@Param("databaseName") String databaseName,
```

Comment:
I'm not sure you need to have the query annotation here. I think `findByDatabaseNameAndTableName(...)` will do what you need it to do automatically.

Reply:
Will it do the filtering by `SCHEDULED` or `FAILED`?

Reply:
Ah fair enough, didn't see that